dht/token.cc

```diff
@@ -79,35 +79,10 @@ sstring token::to_sstring() const {
     return seastar::to_sstring<sstring>(long_token(*this));
 }
 
-// Assuming that x>=y, return the positive difference x-y.
-// The return type is an unsigned type, as the difference may overflow
-// a signed type (e.g., consider very positive x and very negative y).
-template <typename T>
-static std::make_unsigned_t<T> positive_subtract(T x, T y) {
-        return std::make_unsigned_t<T>(x) - std::make_unsigned_t<T>(y);
-}
-
 token token::midpoint(const token& t1, const token& t2) {
-    auto l1 = long_token(t1);
-    auto l2 = long_token(t2);
-    int64_t mid;
-    if (l1 <= l2) {
-        // To find the midpoint, we cannot use the trivial formula (l1+l2)/2
-        // because the addition can overflow the integer. To avoid this
-        // overflow, we first notice that the above formula is equivalent to
-        // l1 + (l2-l1)/2. Now, "l2-l1" can still overflow a signed integer
-        // (e.g., think of a very positive l2 and very negative l1), but
```

Comment:
What about this overflow of l2 - l1 ?

Reply:
I guess l1 can't be negative so we're save.

Reply:
The new code is the same as the true branch of the old code. We still avoid overflow problems by doing the arithmetic in unsigned integers. I simply noticed that the formula for the true branch also works naturally for the false branch.